andy/src/main/java/nl/tudelft/cse1110/andy/execution/step/SourceCodeSecurityCheckStep.java

```diff
@@ -90,14 +92,32 @@ private boolean checkForKeywords(String code, ResultBuilder result) {
         );
         for (String keyword : keywords.keySet()) {
             if (code.contains(keyword)) {
-                result.compilationSecurityFail(keywords.get(keyword));
-                return false;
+                String[] lines = code.split("\\n");
+                // Split lines and search every line for keyword
+                for (int lineNumber = 0; lineNumber < lines.length; lineNumber++)
```

Comment:
Can we have a test showing what this should do? :)

Reply:
A bit of a stupid question but in order to test this should I write unit tests for the three check methods or write integration tests ?